e2e-tests/src/test/adder.rs

```diff
@@ -0,0 +1,71 @@
+use std::str::FromStr;
+
+use aleph_client::{contract::ContractInstance, AccountId, Connection, SignedConnection};
+use anyhow::{Context, Result};
+
+use crate::{
+    config::setup_test,
+    test::helpers::{basic_test_context, sign},
+};
+
+/// This test exercises the aleph-client code for interacting with contracts by testing a simple contract that maintains
+/// some state and publishes some events.
+#[tokio::test]
+pub async fn adder() -> Result<()> {
+    let config = setup_test();
+
+    let (conn, _authority, account) = basic_test_context(config).await?;
+    let contract = AdderInstance::new(
+        &config.test_case_params.adder,
+        &config.test_case_params.adder_metadata,
+    )?;
+
+    let before = contract.get(&conn).await?;
+    contract.add(&sign(&conn, &account), 10).await?;
+    let after = contract.get(&conn).await?;
+
+    assert!(after == before + 10);
+
+    Ok(())
+}
+
+pub(super) struct AdderInstance {
+    contract: ContractInstance,
+}
+
+impl<'a> From<&'a AdderInstance> for &'a ContractInstance {
+    fn from(instance: &'a AdderInstance) -> Self {
+        &instance.contract
+    }
+}
+
+impl<'a> From<&'a AdderInstance> for AccountId {
+    fn from(instance: &'a AdderInstance) -> Self {
+        instance.contract.address().clone()
+    }
+}
```

Comment:
Is it correct that we could implement this for `ContractInstance`:
```rust
impl<'a> From<&'a ContractInstance> for AccountId {
    ...
}
```
and not repeat it for every `ContractInstance`-wrapper

Reply:
I think then you'd need to do `into().into()` at the usage site, which is so-so... I think it would need to be something like `AsRef` or `Deref` to make it seamless. In any case - I would consider those a refactor for later.